libc: minimal: add missing ctype.h functions

cfriedt opened the pull request:
Add the functions below to the minimal libc `ctype.h` since they are missing, and are required as of C89 (C99 for `isblank()`)

* `isblank()`
* `islower()`
* `ispunct()`

Fixes #99452

Commit:
libc: minimal: add missing ctype.h functions

Add the functions below to the minimal libc ctype.h since they are
missing, and are required as of C89 (C99 for `isblank()`)

* `isblank()`
* `islower()`
* `ispunct()`

Signed-off-by: Chris Friedt <cfriedt@tenstorrent.com>

cfriedt commented:
* Added `Compliance: False Positive` because the coding style follows what is currently in the file

keith-packard commented on lib/libc/minimal/include/ctype.h:
```diff
@@ -23,6 +23,11 @@ static inline int isalpha(int c)
+	return (int)((c == (int)' ') || (c == (int)'\t'));
+}
+
 static inline int isspace(int c)
```
I'd let the compiler do something cute if it likes, but express these in a direct fashion.
```
static inline int isspace (int c)
{
    return c == ' ' || ('\t' <= c && c <= '\r');
}
```
Similarly for the other functions. There's no reason to obscure what these functions do; having to check that there are 5 'blank' values between `'\t'` and `'\r'` is something the compiler should do, not us.

cfriedt replied:
I'll prepend a commit for that, since it's technically not related to the bugfix.

keith-packard reviewed (changes requested):
Please express these directly rather than using arithmetic.

Commit:
libc: minimal: ctype: use unsigned int instead of unsigned

The "check warns" workflow in CI warned over the use of `unsigned` as a
shorthand for `unsigned int` in several locations in
`lib/libc/minimal/include/ctype.h`.

```
 UNSPECIFIED_INT: Prefer 'unsigned int' to bare use of 'unsigned'
File:lib/libc/minimal/include/ctype.h
```

Adjust `unsigned` to `unsigned int` to avoid linter warnings.

Signed-off-by: Chris Friedt <cfriedt@tenstorrent.com>

Commit:
libc: minimal: ctype: express ctype limits in more direct fashion

Based on review feedback, it was suggested to express ctype
character checks in a more direct fashion, rather than using
arithmetic, and allow the compiler to optimize as it sees fit.

https://github.com/zephyrproject-rtos/zephyr/pull/99451#\
discussion_r2530339430

Signed-off-by: Chris Friedt <cfriedt@tenstorrent.com>

keith-packard reviewed (changes requested):
Please remove all of the unnecessary casts; C is at least semi-reasonable for these cases.

cfriedt commented on lib/libc/minimal/include/ctype.h:
```diff
@@ -15,42 +15,57 @@ extern "C" {
 static inline int isupper(int a)
 {
-	return (int)(((unsigned)(a)-(unsigned)'A') < 26U);
+	return (int)(((int)'A' <= a) && (a <= (int)'Z'));
```
Ack - yes, that looks like a copy-paste error.

Re casts, I think the casts are there for C++ compilers where a char literal has type char rather than int. It's possible that git / GitHub history might link to a related issue. I'll take a look.

Commit:
libc: minimal: ctype: remove unnecessary casts

A review comment in the PR below requested that unnecessary casts were
removed from ctype.h in the PR below.

https://github.com/zephyrproject-rtos/zephyr/pull/99451

Tested with manual compilation in C and C++ mode with the arguments
```shell
gcc -Wconversion -Werror -Wall -Wextra -Wint-conversion
clang -Wconversion -Werror -Wall -Wextra -Wint-conversion
gcc -Wconversion -Werror -Wall -Wimplicit--Wextra
clang++ -Wconversion -Werror -Wall -Wextra
```

and also with `-- -DCONFIG_COMPILER_WARNINGS_AS_ERRORS=y`

Signed-off-by: Chris Friedt <cfriedt@tenstorrent.com>

aescolar reviewed (commented):
The code is ok, but minimallibc is not meant to be complete
https://docs.zephyrproject.org/latest/develop/languages/c/minimal_libc.html
only to have this subset:
https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#rule-a-4-c-standard-library-usage-restrictions-in-zephyr-kernel
Anyhow, I'm not specially against adding these. 
@stephanosio can decide.
